Scripts/RemediationScripts/Helper.ps1

```diff
@@ -0,0 +1,122 @@
+Set-StrictMode -Version Latest
+
+class ResourceResolver
+{
+	[string[]] $ExcludeResourceNames=@();
+	[PSObject] $ExcludedResources=@();
+	[string[]] $ExcludeResourceGroupNames=@();
+	[string[]] $printMessage=@();
+	
+	ResourceResolver([string] $excludeResourceName , [string] $excludeResourceGroupName)
+	{
+		if(-not [string]::IsNullOrEmpty($excludeResourceName))
+		{
+			$this.ExcludeResourceNames += $this.ConvertToStringArray($excludeResourceName)
+			if ($this.ExcludeResourceNames.Count -eq 0)
+			{
+				throw "The parameter 'ExcludeResourceNames' does not contain any string."
+			}	
+		}
+
+		if(-not [string]::IsNullOrEmpty($excludeResourceGroupName))
+		{
+			$this.ExcludeResourceGroupNames += $this.ConvertToStringArray($excludeResourceGroupName)
+			if ($this.ExcludeResourceGroupNames.Count -eq 0)
+			{
+				throw "The parameter 'ExcludeResourceGroupNames' does not contain any string."
+			}	
+		}
+	}
+
+	[string[]] ConvertToStringArray([string] $stringArray)
+	{
+		$result = @();
+		if(-not [string]::IsNullOrWhiteSpace($stringArray))
+		{
+			$result += $stringArray.Split(',', [StringSplitOptions]::RemoveEmptyEntries) | 
+							Where-Object { -not [string]::IsNullOrWhiteSpace($_) } |
+							ForEach-Object { $_.Trim() } |
+							Select-Object -Unique;
+		}
+		return $result;
+	}
+
+	#Method to filter resources based on exclude flags
+	hidden [PSObject] ApplyResourceFilter([PSobject] $Resources)
+	{	
+		#First remove resource from the RGs specified in -ExcludeResourceGroupNames
+		if(($this.ExcludeResourceGroupNames | Measure-Object).Count -gt 0)
+		{
+			$nonExistingRGS = @()
+			$matchingRGs= $this.ExcludeResourceGroupNames | Where-Object{$_ -in $Resources.ResourceGroupName}
+			$nonExistingRGS += $this.ExcludeResourceGroupNames | Where-Object{$_ -notin $matchingRGs}
+			if(($nonExistingRGS| Measure-Object).Count -gt 0)
+			{
+				#print the message saying these RGS provided in excludeRGS are not found
+				Write-Host "Warning: Did not find the following resource groups requested for exclusion:" -ForegroundColor Yellow
+				Write-Host $($nonExistingRGS -join ", ")
```

Comment:
+ "`n"

Reply:
Warning: Following resource groups requested for exclusion not found in subscription:
1- xyz
2- pqr

Reply:
Added